website/docs/reference/global-configs/about-global-configs.md

```diff
@@ -1,18 +1,97 @@
 
-Global configs control things like the visual output of logs, the manner in which dbt parses your project, and what to do when dbt finds a version mismatch or a failing model. These configs are "global" because they are available for all dbt commands, and because they can be set for all projects running on the same machine or in the same environment.
+Flags control things like the visual output of logs, whether to treat specific warning messages as errors, or whether to "fail fast" after encountering the first error. These flags are "global" because they are available for all dbt commands, and because they can be set in multiple places.
 
-### Global config precedence
+There is significant overlap between dbt's flags and dbt's command line options, but they are not the same:
+- Certain flags can only be set in `dbt_project.yml`, and cannot be overridden for specific invocations via CLI option.
+- If a CLI option is supported by specific commands, rather than supported by all commands ("global"), it is generally not considered to be a "flag".
 
-Starting in v1.0, you can set global configs in three places. dbt will evaluate the configs in the following order:
-1. [user config](https://docs.getdbt.com/reference/global-configs/yaml-configurations)
-1. [environment variable](https://docs.getdbt.com/reference/global-configs/environment-variable-configs)
-1. [CLI flag](https://docs.getdbt.com/reference/global-configs/command-line-flags)
+### Setting flags
 
-Each config is prioritized over the previous one. For example, if all three are provided, then the CLI flag takes precedence.
+There are multiple ways of setting flags, which depend on the use case:
+- **[Project-level `flags` (`dbt_project.yml`)](https://docs.getdbt.com/reference/global-configs/project-flags):** Define version-controlled defaults for everyone running this project. Preserve legacy behaviors until their slated deprecation.
+- **[Environment variables](https://docs.getdbt.com/reference/global-configs/environment-variable-configs):** Define behavior that should be different in different runtime environments (development vs. production vs. [continuous integration](https://docs.getdbt.com/docs/deploy/continuous-integration), or different for different users in development (based on personal preferences).
+- **[CLI options](https://docs.getdbt.com/reference/global-configs/command-line-options):** Define behavior specific to _this invocation_. Supported for all dbt commands.
+
+The most specific setting "wins." If you set the same flag in all three ways, the CLI option will take precedence, followed by the environment variable, followed lastly by the value in `dbt_project.yml`. If you set the flag in none of those places, it will use the default value defined within dbt.
```

Comment:
```suggestion
The most specific setting "wins." If you set the same flag in all three places, the CLI option will take precedence, followed by the environment variable, and finally, the value in `dbt_project.yml`. If you set the flag in none of those places, it will use the default value defined within dbt.
```